moo.js

```diff
@@ -154,6 +190,25 @@
     lineBreaks: true,
   }
 
+  Lexer.prototype.setState = function(state) {
+    if (!state || this.state === state) return
+    this.state = state
+    var index = this.re ? this.re.lastIndex : 0
+    var info = this.states[state]
```

Comment:
Suggestion: warn if state is not known.

Also, wouldn't it make sense for `lexer.setState()` to be an error?